rfc/space-state.md

```diff
@@ -0,0 +1,37 @@
+# Abstract
+
+Currently deployed w3up protocol implementation maintains state of the (user) space in the dynamo db, which is at odds with a notion of user space as they can not alter state locally without active connection to our service.
+
+To resolve this we need to represent state of the (user) space as DAG that user can query and edit locally and sync it with replica held by our service.
+
+## Challenges
+
+### Content Replication
+
+Representing space as a [UnixFS] directory of user content or key value store like [Pail] may seem like a non-brainer, but it introduces major challenge in terms of sync, beacuse:
```

Comment:
Does a user always have only one [UnixFs] directory, or can they have multiple?

Does this map onto s3-like representation? Meaning: does a directory behave like an s3-bucket and blobs are like objects in the bucket?

Reply:
We don't have this, so it's more of a if we used UnixFS dir or Pail this is what the constraints would be. In other words I design of the state DAG is something we need to figure out and then choose the right abstraction for it.

My hypothesis is that we could do it with a flat KV.